Source/Core/InputCommon/ControllerInterface/Xlib/XInput2.cpp

```diff
@@ -131,8 +131,6 @@ void KeyboardMouse::SelectEventsForDevice(Window window, XIEventMask* mask, int
 KeyboardMouse::KeyboardMouse(Window window, int opcode, int pointer, int keyboard)
     : m_window(window), xi_opcode(opcode), pointer_deviceid(pointer), keyboard_deviceid(keyboard)
 {
-  memset(&m_state, 0, sizeof(m_state));
```

Comment:
Any reason you didn't just move this to the declaration and do
```c++
private:
  //...
  State m_state{};
```
in `Xinput2.h` instead?

Reply:
good point. fixed